spec/unit/backtracie_spec.rb

```diff
@@ -271,6 +271,82 @@ class ClassWithBlockInsideMethodDefinedUsingDefinedMethod
+      #     * You are requiring a file, which calls define_method from inside a block. In ruby_memprofiler_pprof,
+      #       the code in question was this, from Bootsnap:
+      #       https://github.com/Shopify/bootsnap/blob/v1.8.1/lib/bootsnap/load_path_cache/change_observer.rb#L49
+      #     * define_method does some allocations, which hit our tracepoint, and we try to capture a backtrace
+      #     * An exception is thrown from inside backtracie: "TypeError: no implicit conversion of nil into String"
+      # It seems, according to my debugger, in this case:
+      #     * The callable_method_entry is Qnil, and
+      #     * The iseq is ISEQ_TYPE_BLOCK
+      # Seems to be a problem in everything from Ruby 2.6 -> 3.1 inclusive (I didn't try anything earlier or later).
+      let(:test_method) do
+        proc do |&blk|
+          bt = nil
+
+          # newobj tracepoints can only be bound from C extensions, but thankfully the issue can be
+          # reproduced with a c-call tracepoint too which has the same effect.
+          set_trace_func(proc do |event, file, line, id, binding, classname|
+            if id == :define_method && event == 'c-call'
+              bt = blk.call
+            end
+          end)
+
+          TOPLEVEL_BINDING.eval <<~RUBY
+            module ToplevelModuleWithDefineMethodInIt
+              %i(some_random_method).each do |m|
+                define_method(m) {}
+              end
+            end
+            Object.send(:remove_const, :ToplevelModuleWithDefineMethodInIt)
+          RUBY
+
+          set_trace_func nil
+          bt
+        end
+      end
+
+      # These two function calls should never be reformatted to be on different lines!
+      # See above for a note on why this looks weird
+      let!(:backtraces_for_comparison) {
+        [test_method.call { described_class.backtrace_locations(Thread.current) }, test_method.call { Thread.current.backtrace_locations }]
+      }
+
+      it_should_behave_like "an equivalent of the Ruby API (using locations)"
+
+      it do
+        expect(backtracie_stack[3].qualified_method_name).to eq "Module\#{block}"
+      end
+    end
+
+    context "when sampling code inside a module" do
+      let(:test_method) do
+        proc do |&blk|
+          $global_block = blk
+          module CapturingBacktraceFromInsideAModule
+            $global_block.call
+          end
+        end
+      end
+
+      # These two function calls should never be reformatted to be on different lines!
+      # See above for a note on why this looks weird
+      let!(:backtraces_for_comparison) {
+        [test_method.call { described_class.backtrace_locations(Thread.current) }, test_method.call { Thread.current.backtrace_locations }]
+      }
+
+      it_should_behave_like "an equivalent of the Ruby API (using locations)"
+
+      it do
+        expect(backtracie_stack[2].qualified_method_name).to eq "Module#<module:CapturingBacktraceFromInsideAModule>"
```

Comment:
This looks kind of ugly but it's also very clear to me what is being called, so I could either change this in the code or leave it how it is. Dealer's choice :)

Reply:
Huh... This is an excellent catch. I totally forgot about this one. 

I had not even given any thought on "what would it be reasonable for this to look like in a stack trace".

Maybe `CapturingBacktraceFromInsideAModule.<definition>` or some madness like it?

For now I think we can live with the version you added. Since we have explicit tests for it, we won't forget :) 

Ahaha Ruby is so interesting :D